bytestring_splitter/__init__.py

```diff
@@ -1,75 +1,113 @@
 
+__all__ = ["__title__", "__summary__", "__version__", "__author__", ]
 
 from contextlib import suppress
 
 VARIABLE_HEADER_LENGTH = 4
 
 
+class BytestringSplittingError(TypeError):
+    """
+    Raised when the bytes don't go in the constructor.
+    """
+
+
 class BytestringSplitter(object):
+    Message = namedtuple("Message", ("name", "message_class", "length", "kwargs"))
+    processed_objects_container = list
 
-    def __init__(self, *message_types_meta):
+    def __init__(self, *message_parameters):
         """
         :param message_types:  A collection of types of messages to parse.
         """
-        self._message_types_meta = message_types_meta
+        if not message_parameters:
+            raise ValueError(
+                "Can't make a BytestringSplitter unless you specify what to split!")
 
-        self.is_variable_length = None
+        self.message_parameters = message_parameters
+        self.message_types = []
+
+        self.is_variable_length = False
         self._length = None
 
-        self.message_types = []
         self._populate_message_types()
 
-        if not message_types_meta:
-            raise ValueError(
-                "Can't make a BytestringSplitter unless you specify what to split!")
-
-        # A quick sanity check here to make sure the message_types don't have a common formatting issue.from
+        # A quick sanity check here to make sure the message_types don't have a common formatting issue.
         # See, you're allowed to pass a simple class and int to make a splitter - that's more-or-less syntactic sugar.
         # You're not allowed to do things like BytestringSplitter((bytes, 3), 17) because what does that even mean?
         with suppress(IndexError):
-            first_message_is_not_int_or_tuple = type(message_types_meta[0]) not in (int, tuple)
-            second_message_is_int = isinstance(message_types_meta[1], int)
+            first_message_is_not_int_or_tuple = type(message_parameters[0]) not in (int, tuple)
+            second_message_is_int = isinstance(message_parameters[1], int)
 
             if first_message_is_not_int_or_tuple and second_message_is_int:
-                raise TypeError("You can't specify the length of the message as a direct argument to the constructor.  Instead, pass it as the second argument in a tuple (with the class as the first argument)")
+                raise TypeError(
+                    "You can't specify the length of the message as a direct argument to the constructor.  Instead, pass it as the second argument in a tuple (with the class as the first argument)")
 
     def __call__(self, splittable, return_remainder=False, msgpack_remainder=False):
 
         if not self.is_variable_length:
             if not any((return_remainder, msgpack_remainder)) and len(self) != len(splittable):
-                raise ValueError(
+                raise BytestringSplittingError(
```

Comment:
`BytestringSplittingError` inherits from `TypeError`, but the error message here (`"Wrong number of bytes ..."`suggests that a `ValueError` should be raised.